services/ui-src/src/components/modals/AddEditReportModal.tsx

```diff
@@ -135,15 +136,44 @@ export const AddEditReportModal = ({
     };
   };
 
+  // NAAAR report payload
+  const prepareNaaarPayload = (formData: any) => {
+    const contactName = formData["contactName"];
+
+    return {
+      metadata: {
+        contactName,
+        lastAlteredBy: full_name,
+        locked: false,
+        submissionCount: 0,
+        previousRevisions: [],
+        naaarReport,
+      },
+      fieldData: {
+        contactName,
+      },
+    };
+  };
+
   const writeReport = async (formData: any) => {
     setSubmitting(true);
     const submitButton = document.querySelector("[form=" + form.id + "]");
     submitButton?.setAttribute("disabled", "true");
 
-    const dataToWrite =
-      reportType === "MCPAR"
-        ? prepareMcparPayload(formData)
-        : prepareMlrPayload(formData);
+    let dataToWrite;
+    switch (reportType) {
+      case "MCPAR":
+        dataToWrite = prepareMcparPayload(formData);
+        break;
+      case "MLR":
+        dataToWrite = prepareMlrPayload(formData);
+        break;
+      case "NAAAR":
+        dataToWrite = prepareNaaarPayload(formData);
+        break;
+      default:
+        throw new Error("Issue creating report");
```

Comment:
Will this line ever get hit? Does a form render if the `reportType` doesn't match one of those defined here? https://github.com/Enterprise-CMCS/macpro-mdct-mcr/blob/ab2830fa73c8b320cb629f1a3f17877caf72a746/services/ui-src/src/components/modals/AddEditReportModal.tsx#L46-L50

Unless a form renders, I don't think this method would get called with a bad `reportType`.

Reply:
I think you are right. Without the default throwing an error later references to `dataToWrite` popup typescript errors because it might be undefined. do you think there is a good way of defining it to avoid that? or maybe just do a check after the switch to ensure it got defined? not sure

Reply:
What about setting the default as `prepareMlrPayload(formData)` as the previous code did?